### PR TITLE
DSD-1871: Add item count to MultiSelect

### PR DESCRIPTION
**Pull request by jackiequach**
Fixes JIRA ticket [DSD-1871](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1871)

## This PR does the following:

- Adds optional `itemCount` data property to the `items` prop of `MultiSelect` to render the item count for the associated option.

## How has this been tested?

- Locally on Storybook

## Accessibility concerns or updates

- N/A

### Accessibility Checklist

- [ ] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [ ] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [ ] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [ ] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [ ] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

-

### Checklist:

- [ ] I have updated the Storybook documentation and changelog accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

[DSD-1871]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1871?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/MultiSelect/MultiSelect.tsx [hunk 1]
```diff
@@ -303,6 +305,19 @@ export const MultiSelect: ChakraComponent<
         );
       };
 
+      const getItemLabelText = (
+        item: MultiSelectItem
+      ): string | JSX.Element => {
+        return item.itemCount ? (
+          <Flex justify="space-between">
```
Comment (bigfishdesign13) [inline src/components/MultiSelect/MultiSelect.tsx]: Add `gap="s"`.

File: src/components/MultiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -18,6 +19,7 @@ export interface MultiSelectItem {
   name: string;
   isDisabled?: boolean;
   children?: MultiSelectItem[];
+  itemCount?: number;
```
Comment (bigfishdesign13) [inline src/components/MultiSelect/MultiSelect.tsx:22]: I like that this will not allow something like `"8 items"` to be passed.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
This is great. It just needs one minor cosmetic update.

**Review by bigfishdesign13 — APPROVED** [review]
Looks great!

**Review by EdwinGuzman — COMMENTED** [review]
This works but before approving, did you try typing the `name` property in `MultiSelectItem` as `string | JSX.Element`? That's what I thought the approach would be since it's the same approach as the `Checkbox` component. That lets the consuming app handle pass the structured DOM themselves so the DS component is not responsible.

This approach give us more of the control which is okay, but just wondering if you tried the other approach and it didn't work.

**Comment by jackiequach** [comment]
@EdwinGuzman I did not try the other approach. I think I misunderstood our conversation during planning. I can implement the other approach for consistency.

**Comment by EdwinGuzman** [comment]
@jackiequach no worries if you haven't already started doing so. This is a good approach and does make it easier (perhaps?) for consuming devs to pass in an additional property instead of managing the DOM themselves.

**Review by EdwinGuzman — APPROVED** [review]
Approved but don't merge just yet.

**Pull request merged**